src/main/java/com/synopsys/integration/alert/web/api/role/RoleActions.java

```diff
@@ -56,19 +55,17 @@
 @Component
 public class RoleActions {
     private static final String FIELD_KEY_ROLE_NAME = "roleName";
-    private final RoleAccessor roleAccessor;
     private final AuthorizationManager authorizationManager;
```

Comment:
I'm not sure delegating Role retrieval to the AuthorizationManager is the right move. Yes, AuthorizationManager has a RoleAccessor, and yes, the RoleActions would otherwise need both, but I do not believe this is correctly separating concerns. RoleAccessor is the thing that should be depended on to read/write roles to and from the database and AuthorizationManager is the utility that cross-references roles with users.

I'm not entirely sure why AuthorizationManager was even being used in this class before.

Reply:
The AuthorizationManager is being used to make sure the cache is kept up to date.

I was not sure about adding the get's to AuthorizationManager. I would be fine with rejecting this PR, if the team thinks it would be better to keep the role retrieval in RoleAccessor.

Reply:
That's my preference, but I'm also second guessing the permissions cache. I can't remember the motivation for it. It seems to me that, if anything, there should be an object representing just the cache that gets used only when needed.